hs_core/page_processors.py

```diff
@@ -311,7 +336,12 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
                                               type_value != 'hasPart'),
                'show_web_reference_note': has_web_ref,
                'belongs_to_collections': belongs_to_collections,
-               'maps_key': maps_key
+               'maps_key': maps_key,
+               'communities_enabled': settings.COMMUNITIES_ENABLED,
+               'topics_json': mark_safe(escapejs(json.dumps(topics))),
+               'admin_user': True,  # TODO this is probably no longer needed since the link to topics editing was moved to admin tab of community
+               'czo_user': any("CZO National" in x.name for x in comms_member_of),
+               'odm2_terms': list(ODM2Variable.all())
```

Comment:
This is the only places ODM2Variable is used and it's passed to the frontend, no logic in the backend.  Couldn't you just query ODM2 as needed on the frontend instead of maintaining a copy in HydroShare?  I don't see how a copy is helping us and it's just more to maintain.

Reply:
We should have a call with @alvacouch and @horsburgh about the long-term benefits and plans to incorporate parts of ODM2 into HydroShare. Some of the new features you see in "CZO" work are just beta for things that (with modification) can be nicely modified for the entire HS user base.

Reply:
Yeah, I guess I'll have to know more about future plans with ODM2, but right now, we're just creating work for ourselves by syncing in they hydroshare backend.  The call to get the list of ODM2Variables could easily be done on the frontend as things stand today.

Reply:
Let's discuss more with Alva and Jeff.

Reply:
@sblack-usu the intent here is to cache ODM2 terms, to guard against performance and uptime issues that could arise in the external ODM2 service. Can you sync up with Jeff on this?

Reply:
Funny that you should mention this today as we are having an outage of part of our virtualization system that is affecting the vocabulary.odm2.org website.

Reply:
All of the Critical Zone stakeholders are reviewing the demo and there is a critical meeting today, reviewing the UI. This would have been a severely negative impact to the project if we weren't caching the ODM2 here. Scott, if you're ok with this, let's mark this as Resolved in GH `Resolve conversation`

Reply:
Not saying the caching is the right solution or the full solution, just that the vocabulary website is down for the moment and I'm working to get it back. The vocabularies were meant to be dynamic and so any cache would also have to be dynamic (updated frequently).

Reply:
Scott and I had a conversation and agree that a separate document/process should be maintained to track 'what will be accepted in the PR and merged to dev' and 'what items will change and in what timeframe'. Decisions that impact change might be 'future communities, generalization of what a community represents and what is needed in the UI' as well as ongoing supporting systems and infrastructure changes, such as ODM2

Reply:
There is a performance hit to an external query that is not present for an internal query. This is why I ingested it. Long-term, this will allow auto-complete on ODM2 terms when specifying keywords. This was the intent of ingesting the database rather than querying it.